executor/index_merge_reader_test.go

```diff
@@ -50,3 +50,14 @@ func (s *testSuite1) TestJoin(c *C) {
 	tk.MustQuery("select /*+ use_index_merge(t1, t1a, t1b) */ sum(t1.a) from t1 join t2 on t1.id = t2.id where t1.a < 2 or t1.b > 4").Check(testkit.Rows("6"))
 	tk.MustQuery("select /*+ use_index_merge(t1, t1a, t1b) */ sum(t1.a) from t1 join t2 on t1.id = t2.id where t1.a < 2 or t1.b > 5").Check(testkit.Rows("1"))
 }
+
+func (s *testSuite1) TestIndexMergeReaderAndGeneratedColumn(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
```

Comment:
Do we need to `	tk.MustExec("set tidb_enable_index_merge=1")` ?

Reply:
No, the hint doesn't require `tidb_enable_index_merge`.

Reply:
ok, but I think it's a bug.

If we do not enable_index_merge, the hint should not take effect

Reply:
Yes, I think so.

Reply:
I don't want to interfere, but the [documentation](https://pingcap.github.io/docs/dev/reference/performance/index-merge/#enable-indexmerge) currently mentions the following:

> IndexMerge is disabled by default. Enable the IndexMerge in one of two ways:
> * Set the tidb_enable_index_merge system variable to 1;
> * Use the SQL Hint USE_INDEX_MERGE in the query.
> 
> Note: The SQL Hint has a higher priority over the system variable.

Reply:
Alright 😢